### PR TITLE
Move Sha256Hash and other Hash algorithm types from KeyVault to Azure::Core

### PR DESCRIPTION
**Pull request by ahsonkhan**
First review https://github.com/Azure/azure-sdk-for-cpp/pull/2523 and https://github.com/Azure/azure-sdk-for-cpp/pull/2525

Relevant commit to review: https://github.com/Azure/azure-sdk-for-cpp/commit/4482006c6da7e7ad83ac7b7a3d56cf451e8b66a4

Third and last step of of https://github.com/Azure/azure-sdk-for-cpp/issues/2500#issuecomment-871761727

### CODE REVIEW

**Review by antkmsft — APPROVED** [review]

**Review by Jinming-Hu — APPROVED** [review]

**Review by vhvb1989 — APPROVED** [review]
This will conflict too much with the other renaming PR in the queue.

**Comment by ahsonkhan** [comment]
> This will conflict too much with the other renaming PR in the queue.

@vhvb1989 fixed, please let me know if you find any issues during the review :)

File: sdk/keyvault/azure-security-keyvault-common/CMakeLists.txt [hunk 1]
```diff
@@ -30,16 +30,13 @@ set(
   AZURE_KEYVAULT_COMMON_HEADER
     inc/azure/keyvault/common/internal/keyvault_pipeline.hpp
     inc/azure/keyvault/common/internal/single_page.hpp
-    inc/azure/keyvault/common/internal/sha_hash.hpp
 )
 
 set(
   AZURE_KEYVAULT_COMMON_SOURCE
+    src/keyvault_pipeline.cpp
```
Comment (antkmsft) [inline sdk/keyvault/azure-security-keyvault-common/CMakeLists.txt:37]: nit: I think it is better when directories go first.

**Pull request merged**